File: android/app/src/main/java/com/dismoi/scout/Floating/Layout/Message.java
package com.dismoi.scout.Floating.Layout;

import android.content.Context;
import android.util.AttributeSet;
import android.view.WindowManager;
import android.animation.AnimatorInflater;
import android.animation.AnimatorSet;
import android.animation.ValueAnimator;
import android.os.Handler;
import android.os.Looper;

import android.util.Log;

import com.dismoi.scout.R;

public class Message extends Layout {
  private final WindowManager windowManager;
  private final MoveAnimator animator;
  private Context c;
  private int width;

  public Message(Context context, AttributeSet attrs) {
    super(context, attrs);

    Log.v("Notifications", "Message view animator");

    animator = new MoveAnimator();
    c = context;
    windowManager = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
    initializeView();
  }

  private void initializeView() {

    Log.v("Notifications", "initialize view");

    setClickable(true);
  }

  @Override
  protected void onAttachedToWindow() {

    Log.v("Notifications", "on attached to window");

    super.onAttachedToWindow();
    playAnimation();
  }

  private void playAnimation() {

    Log.v("Notifications", "play animation");
    AnimatorSet animator = (AnimatorSet) AnimatorInflater
              .loadAnimator(getContext(), R.animator.message_slide_up);
    animator.setTarget(this);
    animator.start();
  }

  private void move(float deltaX, float deltaY) {

    Log.v("Notifications", "move");

    getViewParams().x += deltaX;
    getViewParams().y += deltaY;
    windowManager.updateViewLayout(this, getViewParams());
  }

  private class MoveAnimator implements Runnable {
    private final Handler handler = new Handler(Looper.getMainLooper());
    private float destinationX;
    private float destinationY;
    private long startingTime;

    private void start(float x, float y) {

      Log.v("Notifications", "start");

      this.destinationX = x;
      this.destinationY = y;
      startingTime = System.currentTimeMillis();
      handler.post(this);
    }

    @Override
    public void run() {
      Log.v("Notifications", "run");
      if (getRootView() != null && getRootView().getParent() != null) {
        float progress = Math.min(1, (System.currentTimeMillis() - startingTime) / 400f);
        float deltaX = (destinationX -  getViewParams().x) * progress;
        float deltaY = (destinationY -  getViewParams().y) * progress;
        move(deltaX, deltaY);
        if (progress < 1) {
          handler.post(this);
        }
      }
    }

    private void stop() {
      Log.v("Notifications", "stop");
      handler.removeCallbacks(this);
    }
  }

}

File: android/app/src/main/java/com/dismoi/scout/floating/layout/Message.kt
package com.dismoi.scout.floating.layout

import android.content.Context
import android.os.Handler
import android.os.Looper
import android.util.AttributeSet
import android.util.Log
import android.view.WindowManager
import com.dismoi.scout.R

import android.animation.*
import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.view.animation.AccelerateInterpolator
import android.view.animation.LinearInterpolator
import android.widget.Button
import android.widget.FrameLayout
import android.widget.ImageView
import androidx.appcompat.widget.AppCompatImageView

class Message(context: Context, attrs: AttributeSet?) : Layout(context, attrs) {

  lateinit var star: ImageView
  lateinit var rotateButton: Button
  lateinit var translateButton: Button
  lateinit var scaleButton: Button
  lateinit var fadeButton: Button
  lateinit var colorizeButton: Button
  lateinit var showerButton: Button

  override fun onAttachedToWindow() {
    Log.d("Notification", "Hello")
    super.onAttachedToWindow()
    playAnimation()
  }

  private fun playAnimation() {
    val animator = ObjectAnimator.ofFloat(this, View.TRANSLATION_Y, 600f, 0f)

    animator.start()
  }
}
